includes/Handlers/Order.php

```diff
@@ -1052,4 +1053,22 @@ public function no_verification_on_guest_save( $email_verification_required ) {
 
 		return $email_verification_required;
 	}
+
+	/**
+	 * Loads the class `WC_Order_Square` to handle deprecated
+	 * dynamic properties notice for PHP 8.2+.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param string $classname Order classname.
+	 *
+	 * @return string
+	 */
+	public function load_custom_order_class( $classname ) {
+		if ( '\Automattic\WooCommerce\Admin\Overrides\Order' === $classname ) {
```

Comment:
@woocommerce/quark requesting you to review this approach taken to resolve the dynamic properties notice.

Reply:
hmm it's not ideal. Seems like a bit of an arms race and potentially flaky if other folks need to do something similar. 

For example, WC core are already doing this for some purpose I don't fully understand but it seems to add functionality that might break if it's overridden further.

Having said that, I don't know if there's an alternative other than a rethink of how the Square plugin uses these dynamic properties given the current environment. From the discussion on this issue (https://github.com/woocommerce/woocommerce/issues/31316) and this PR (https://github.com/woocommerce/woocommerce/issues/35763), they don't seem open to broadly rolling out the `AllowDynamicProperties` flag across the core WC classes.

Reply:
@diegocurbelo what's your thoughts on this then, acceptable as-is or someone else to consult on most optimal approach here?

Reply:
Note that this just resolves some deprecated notices in PHP 8.2+ so if we're unsure on this approach, we could remove this for now and come back later.

Eventually those deprecated notices will be warnings or errors I imagine so something that needs fixed but isn't the most crucial thing at the moment.

Reply:
Hmm, interesting discussion on those WooCommerce issues and I agree with @james-allan that this change isn't ideal... I ultimately don't see an alternative to this problem though and the workarounds others shared on those issues don't seem ideal either 🤔 

The only suggestion I was going to make was rather than filtering every single `woocommerce_order_class` hook, could we get away with limiting this to just our `get_order()`, `get_order_for_capture()` and `get_order_for_refund()` functions to use `WC_Order_Square` when we need it?


i.e. 

```
if ( is_numeric( $order ) ) {
	/** @var WC_Order_Square $order */
	add_filter( 'woocommerce_order_class', array( $this, 'load_custom_order_class' ), 999 );
	$order = wc_get_order( $order );
	remove_filter( 'woocommerce_order_class', array( $this, 'load_custom_order_class' ), 999 );
}
```

Or would this potentially introduce other problems?

Reply:
> Or would this potentially introduce other problems?

I don't have an answer for this 😓. We can give this a try but it will be tough to imagine for what particular configuration(s) this can fail. 

@jeffpaul there is still an option to refactor the plugin which will be some decent amount of work, both development and testing wise. Would you suggest we take this approach?

Reply:
@jeffpaul @dkotter any suggestions on the approach noted by Sid above?